harvest/api/alpaca.py

```diff
@@ -0,0 +1,275 @@
+# Builtins
+import yaml
+import datetime as dt
+from logging import critical, error, info, warning, debug
+from typing import Any, Dict, List, Tuple
+
+# External libraries
+import pandas as pd
+from alpaca_trade_api.rest import REST, TimeFrame
+
+# Submodule imports
+from harvest.api._base import API
+from harvest.utils import *
+
+class Alpaca(API):
+
+    def __init__(self, path: str=None, is_basic_account: bool=False, paper_trader: bool=False):
+        super().__init__(path)
+        self.basic = is_basic_account
+
+        endpoint = 'https://paper-api.alpaca.markets' if paper_trader else 'https://api.alpaca.markets'
+        self.api = REST(self.config.api_key, self.config.secret_key, endpoint)
+
+    def no_secret(self, path: str) -> bool:
+        return self.create_secret(path)
+
+    def setup(self, watch: List[str], interval: str, trader=None, trader_main=None):
+        self.watch_stock = []
+        self.watch_crypto = []
+
+        for s in watch:
+            if is_crypto(s):
+                self.watch_crypto.append(s)
+            else:
+                self.watch_stock.append(s)
+
+        self.option_cache = {}
+        super().setup(watch, interval, interval, trader, trader_main)
+
+    def exit(self):
+        self.option_cache = {}
+
+    def main(self):
```

Comment:
Alpaca offers a streaming API, so it is better to use that instead of polling.

Reply:
Now uses the streaming service to get the latest price. Note that I think it updates per minute so if someone sets the interval to 1HR they will still get the price per minute and not aggregated over an hour.